Source/WebCore/platform/graphics/gstreamer/ImageGStreamerSkia.cpp

```diff
@@ -91,17 +95,24 @@ ImageGStreamer::ImageGStreamer(GRefPtr<GstSample>&& sample)
         break;
     }
 
+    m_size = { static_cast<float>(videoFrame->width()), static_cast<float>(videoFrame->height()) };
+
     auto toSkiaColorSpace = [](const PlatformVideoColorSpace&) {
         notImplemented();
         return SkColorSpace::MakeSRGB();
     };
-    auto imageInfo = SkImageInfo::Make(videoFrame->width(), videoFrame->height(), colorType, alphaType, toSkiaColorSpace(videoColorSpaceFromInfo(videoInfo)));
-    SkPixmap pixmap(imageInfo, videoFrame->planeData(0), videoFrame->planeStride(0));
-    auto image = SkImages::RasterFromPixmap(pixmap, [](const void*, void* context) {
-        std::unique_ptr<GstMappedFrame> videoFrame(static_cast<GstMappedFrame*>(context));
-    }, videoFrame.release());
-
-    m_image = BitmapImage::create(WTFMove(image));
+    auto imageInfo = SkImageInfo::Make(videoFrame->width(), videoFrame->height(), colorType, alphaType, toSkiaColorSpace(videoColorSpaceFromInfo(*videoInfo)));
+
+    // Copy the buffer data. Keeping the whole mapped GstVideoFrame alive would increase memory
+    // pressure and the file descriptor(s) associated with the buffer pool open. We only need the
+    // data here.
+    auto bufferData = createBufferData();
+    bufferData->planeData = { reinterpret_cast<const uint8_t*>(videoFrame->planeData(0)), static_cast<unsigned>(videoFrame->planeStride(0)) };
+
+    SkPixmap pixmap(imageInfo, bufferData->planeData.data(), bufferData->planeData.size_bytes());
+    m_image = SkImages::RasterFromPixmap(pixmap, [](const void*, void* context) {
+        destroyBufferData(reinterpret_cast<BufferData*>(context));
+    }, bufferData);
```

Comment:
Is it really needed to copy the data? If it is, it should be simpler to use RasterFromPixmapCopy so that skia copies the data and takes the ownership, so you don't need the async struct thing.